### DISCUSSION TITLE
Error: Assertion failed: Expected the resolution for webpack@npm:^5.54.0 [ef0ec] to have been registered

### DISCUSSION THREAD

**Discussion by fabis94** [yarnpkg/berry#4704] [Q&A]
Any idea what circumstances cause this kind of error to happen? I have a Vue project w/ Vue CLI and some plugins that I'm trying to upgrade from v4 to v5, but I'm getting this error and it's not exactly obvious what to do here. Couldn't find anything at all about this in Google either. So how should I debug this issue further?

Reproduction link: https://github.com/specklesystems/speckle-server/pull/888/files
Just check out that branch and do `yarn`

It started happening as I moved the webpack, @vue/cli-* dependencies up to latest versions in the `frontend` package.json

**Comment by HosseinAgha** [comment]
I was encountering the exact same error for `styled-components`.  
My project structure was like this

workspace a -> workspace b
workspace b -> styled-components

and a resolutions field for `styled-components` in root.  

The issue was that I had to add `styled-components` to `workspace a` dependencies. No other way around it.

**Reply by HosseinAgha** [reply]
I'm encountering this issue again.  
After the above fix, If I try to add any new module this error happens again.

**Comment by ShravanSunder** [comment]
Also having this issue with a monorepo importing hardhat deploy "@nomiclabs/hardhat-ethers": "npm:hardhat-deploy-ethers@^0.3.0-beta.13",

**Comment by vishaldodiya** [comment]
I am also having this issue in my monorepo setup. `Error: Assertion failed: Expected the resolution for react-router-dom@npm:^5.1.0 [0ed99] to have been registered`
